client/core/core.go

```diff
@@ -2682,6 +2687,42 @@ func (c *Core) walletIsActive(assetID uint32) bool {
 	return false
 }
 
+func (dc *dexConnection) bondOpts() (assetID uint32, targetTier, max uint64) {
+	dc.acct.authMtx.RLock()
+	defer dc.acct.authMtx.RUnlock()
+	return dc.acct.bondAsset, dc.acct.targetTier, dc.acct.maxBondedAmt
+}
+
+func (dc *dexConnection) hasActiveBond(assetID uint32) bool {
+	check := func(bonds []*db.Bond) bool {
+		for _, b := range bonds {
+			if assetID == b.AssetID {
+				return true
+			}
+		}
+		return false
+	}
+	dc.acct.authMtx.RLock()
+	defer dc.acct.authMtx.RUnlock()
+	return check(dc.acct.bonds) || check(dc.acct.pendingBonds) || check(dc.acct.expiredBonds)
+}
+
+// isActiveBondAsset indicates if a wallet is needed for bonding on any
+// configured DEX. includeLive should be set to consider all existing unspent
+// bonds that need to be refunded in the future.
+func (c *Core) isActiveBondAsset(assetID uint32, includeLive bool) bool {
+	for _, dc := range c.dexConnections() {
+		bondAsset, targetTier, _ := dc.bondOpts()
+		if bondAsset != assetID {
+			continue
```

Comment:
Although refunds do not require an unlocked wallet (the private key is stored in the `db.Bond`), it would certainly not be ok to *disable* a wallet that has active bonds to refund.   The `includeLive` check below is intended to check that case, but now I see from your comment that the `continue` above will preclude that check unintentionally.  Will fix.